e2e/tests-dfx/request_status.bash

```diff
@@ -29,3 +29,23 @@ teardown() {
   assert_eq '4449444c0001710b48656c6c6f2c20426f6221'
 
 }
+
+@test "request-status requires same identity" {
+  install_asset greet
+  dfx_start --artificial-delay 10000
```

Comment:
Great question! I copied this from the test above.  The delays were introduced in https://github.com/dfinity/sdk/pull/3920. @lwshang , do you remember?

Reply:
Later in this test, we try to make a call asynchronously and get the request-id (in stdout): 
```
dfx canister call --async hello_backend greet Bob
```
Then we request-status of the request
```
dfx canister request-status --output raw "$stdout" "$(dfx canister id hello_backend)"
```

In my PR #3920, I bumped the agent-rs dependency to v0.38 which contains this change:  https://github.com/dfinity/agent-rs/pull/595

Without the `--artificial-delay 10000`, the first step won't output the request-id, instead it will output the call result `("Hello, Bob!")` directly.
Adding `--artificial-delay 10000` to `dfx start` served as a workaround to preserve the existing test behavior.